src/pygeoprocessing/routing/routing.pyx

```diff
@@ -2960,3 +3006,1298 @@ def _is_raster_path_band_formatted(raster_path_band):
+                    f'{x_l}, {y_l} out of bounds for '
+                    f'{n_cols}x{n_rows} raster.')
+            if edge_side - ((edge_dir-2) % 8) == 0:
+                # counterclockwise configuration
+                left = edge_dir
+                right = (left-1) % 8
+                out_dir_increase = 2
+            else:
+                # clockwise configuration (swapping "left" and "right")
+                right = edge_dir
+                left = (edge_side+1)
+                out_dir_increase = -2
+            pixel_move = 1
+            left_in = _in_watershed(
+                x_l, y_l, left, discovery, finish, n_cols, n_rows,
+                discovery_managed_raster, discovery_nodata)
+            right_in = _in_watershed(
+                x_l, y_l, right, discovery, finish, n_cols, n_rows,
+                discovery_managed_raster, discovery_nodata)
+            if right_in:
+                # turn right
+                out_dir = edge_side
+                edge_side = (edge_side-out_dir_increase) % 8
+                edge_dir = out_dir
+                # pixel moves to be the right cell
+                x_l += D8_XOFFSET[right]
+                y_l += D8_YOFFSET[right]
+                _diagonal_fill_step(
+                    x_l, y_l, right,
+                    discovery, finish, discovery_managed_raster,
+                    discovery_nodata,
+                    boundary_list)
+                n_left_turns = 0
+            elif left_in:
+                # step forward
+                x_l += D8_XOFFSET[edge_dir]
+                y_l += D8_YOFFSET[edge_dir]
+                # the pixel moves forward
+                boundary_list.append((x_l, y_l))
+                n_left_turns = 0
+            else:
+                # turn left
+                edge_side = edge_dir
+                edge_dir = (edge_side + out_dir_increase) % 8
+                n_left_turns += 1
+
+            if _is_close(x_p, x_first) and _is_close(y_p, y_first):
+                # met the start point so we completed the watershed loop
+                break
+
+        watershed_feature = ogr.Feature(watershed_layer.GetLayerDefn())
+        watershed_polygon = ogr.Geometry(ogr.wkbPolygon)
+        watershed_polygon.AddGeometry(watershed_boundary)
+        watershed_feature.SetGeometry(watershed_polygon)
+        watershed_feature.SetField('stream_fid', stream_fid)
+        watershed_layer.CreateFeature(watershed_feature)
+
+        # this loop fills in the raster at the boundary, done at end so it
+        # doesn't interfere with the loop return to think the cells are no
+        # longer in the watershed
+        for boundary_x, boundary_y in boundary_list:
+            discovery_managed_raster.set(boundary_x, boundary_y, -1)
+    watershed_layer.CommitTransaction()
+    watershed_layer = None
+    watershed_vector = None
+    discovery_managed_raster.close()
+    finish_managed_raster.close()
+    #shutil.rmtree(workspace_dir)

```

Comment:
Lingering deprecated code?

Reply:
So weird, that line is not in my working file, but otherwise yes, I would have removed it.